src/device/wg_controller.rs

```diff
@@ -191,27 +191,42 @@ async fn consolidate_wg_peers<
                 }
 
                 // Initiate session keeper to start sending data between peers connected directly
-                if let (Some(sk), Some(mesh_ip)) =
-                    (session_keeper, requested_peer.peer.allowed_ips.first())
-                {
-                    // TODO this should be acommodatedto IPv6
-                    let ip4 = {
-                        match mesh_ip.ip() {
-                            IpAddr::V4(ip4) => ip4,
-                            IpAddr::V6(_ip6) => {
-                                telio_log_error!(
-                                    "IPv6 not supported, session keeper will not run on peer {:?}",
-                                    requested_peer.peer.public_key
-                                );
-                                break;
+                if let (Some(sk), Some(mesh_ip1), maybe_mesh_ip2) = (
+                    session_keeper,
+                    requested_peer.peer.allowed_ips.get(0),
+                    requested_peer.peer.allowed_ips.get(1),
+                ) {
+                    let target = {
+                        let mut ret: SessionKeeperTarget = (None, None);
+
+                        match mesh_ip1.ip() {
+                            IpAddr::V4(ip4) => ret.0 = Some(ip4),
+                            IpAddr::V6(ip6) => ret.1 = Some(ip6),
+                        }
+
+                        if let Some(mesh_ip2) = maybe_mesh_ip2 {
+                            match mesh_ip2.ip() {
+                                // If both addrs are from same protocol (v4, v4) or (v6, v6), we're taking only the first one received.
+                                IpAddr::V4(ip4) => {
+                                    if ret.0.is_none() {
+                                        ret.0 = Some(ip4)
+                                    }
+                                }
+                                IpAddr::V6(ip6) => {
+                                    if ret.1.is_none() {
+                                        ret.1 = Some(ip6)
+                                    }
+                                }
                             }
                         }
```

Comment:
```suggestion
                        match maybe_mesh_ip2.map(|ip| ip.ip()) {
                            // If both addrs are from same protocol (v4, v4) or (v6, v6), we're taking only the first one received.
                            Some(IpAddr::V4(ip4)) => {
                                if ret.0.is_none() {
                                    ret.0 = Some(ip4)
                                 }
                            }
                            Some(IpAddr::V6(ip6)) => {
                                if ret.1.is_none() {
                                    ret.1 = Some(ip6)
                                 }
                             }
                            None => {}
                         }
```


You can remove one layer of nesting by using `.map(..)`, Maybe it is a bit simpler this way?

Reply:
Or maybe even this:

```suggestion
let ret: SessionKeeperTarget = match (mesh_ip1.ip(), maybe_mesh_ip2.map(|ip| ip.ip())) { : &IpNetwork                                                          
    (IpAddr::V4(ip4), Some(IpAddr::V6(ip6))) => (Some(ip4), Some(ip6)), : Ipv4Addr : Ipv6Addr                                                                  
    (IpAddr::V6(ip6), Some(IpAddr::V4(ip4))) => (Some(ip4), Some(ip6)), : Ipv6Addr : Ipv4Addr                                                                        
    (IpAddr::V4(ip4), _) => (Some(ip4), None), : Ipv4Addr                                                                                                      
    (IpAddr::V6(ip6), _) => (None, Some(ip6)), : Ipv6Addr                                                                                                      
};    
```

Reply:
But even better would be to refactor SessionKeeper to not accept primary and secondary addresses as IpAddr instead of specific Ipv4Addr or Ipv6Addr, and then just control ipv4 vs ipv6 within the validation itself, this maybe would allow wg_controller to be ip-version agnostic by simply using IpAddr structs.

Reply:
Fixed!

Reply:
I would guess, that we won't supply `SessionKeeper` with two IPv6 or two IPv4 addresses, so input logic as it is now, satisfy our needs.